spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/ConfigMapConfigProperties.java

```diff
@@ -98,6 +109,17 @@ public String getConfigurationTarget() {
 		 */
 		private String namespace;
 
+		/**
+		 * Use config map name as prefix for properties. Can't be a primitive, we need to
+		 * know if it was explicitly set or not
+		 */
+		private Boolean useNameAsPrefix;
```

Comment:
so at the level of : `spring.cloud.kubernetes.config.useNameAsPrefix`, having a primitive `boolean` is fine; but having the same here might mess up with a configuration. For example, let's suppose we have it as primitive here also. Default value for it is `false`. So imagine I do this:

```
spring:
  application:
    name: with-prefix
  cloud:
    kubernetes:
      config:
        useNameAsPrefix: true
        namespace: default-namespace
        sources:
          - name: config-map-one
          - name: config-map-two
            useNameAsPrefix: false
```

Obviously, having a primitive, will result in the fact that  `config-map-one` will not use its name as prefix, when it really should.